v4/newrelic/segments_test.go

```diff
@@ -740,31 +744,49 @@ func TestSegmentsAddAttribute(t *testing.T) {
 		End()
 	}
 
-	testFns := []func(*Transaction) segment{
-		func(txn *Transaction) segment {
-			return txn.StartSegment("basic")
+	testcases := []struct {
+		start      func(*Transaction) segment
+		extraAttrs map[string]interface{}
+	}{
+		{
+			start: func(txn *Transaction) segment {
+				return txn.StartSegment("basic")
+			},
 		},
-		func(txn *Transaction) segment {
-			return &DatastoreSegment{
-				StartTime: txn.StartSegmentNow(),
-			}
+		{
+			start: func(txn *Transaction) segment {
+				return &DatastoreSegment{
+					StartTime: txn.StartSegmentNow(),
+				}
+			},
+			extraAttrs: map[string]interface{}{
+				"db.system":     "unknown",
+				"db.statement":  "'unknown' on 'unknown' using 'unknown'",
+				"db.operation":  "unknown",
+				"net.peer.name": "unknown",
+				"db.name":       "unknown",
+			},
```

Comment:
This block of changes needed to happen because there's now more attributes on some spans that we need to account for. This test is looking for custom attributes on spans. But since the custom attrs and the attrs that our shim makes are all in the same bucket, we must also look for these attrs made by the shim as well.